test.sh

```diff
@@ -11,37 +11,57 @@ This script creates virtual network adapter which can be used for bridging
 
 Usage:
 $0 [options]
-
 Options:
-    start   This option will create virtual adpater
-    stop    This option will destory virtual adpater
+    start   This option will create virtual adapter
+    stop    This option will destory virtual adapter
     
 _EOF
 }
 
-if [[ $UID != 0 ]]; then
-	echo "error: use sudo or run script as root user"
-	exit 1
-fi
-
 case $1 in
     start )
+        
         echo "creating virtual interface.."
-        modprobe dummy
-        ip link add ${DEVICE} type dummy
-        ip link set ${DEVICE} up
-        ip addr add ${NETWORK}.100/24 brd + dev ${DEVICE}
+        sudo modprobe dummy
+        sudo ip link add ${DEVICE} type dummy
+        sudo ip link set ${DEVICE} up
+        sudo ip addr add ${NETWORK}.100/24 brd + dev ${DEVICE}
         sed -i settings.sh -e "/^LAN_IF/ c LAN_IF=\"${DEVICE}\"" 
+        
+        vagrant destroy -f
+        cd dhcp
```

Comment:
`pushd` dhcp and `popd` is better alternative. No need to change, but just for info.

Reply:
Nice.